### PR TITLE
Add support for Puppet 7

### PR DESCRIPTION
**Pull request by smortex**
This PR also include:

* #1126 
* #1127

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -53,31 +53,26 @@
...
         "18.04",
         "20.04"
       ]
     },
     {
       "operatingsystem": "SLES",
       "operatingsystemrelease": [
-        "12.1",
-        "12.2"
       ]
```
Comment (bastelfreak) [inline metadata.json:69]: I thought SLES still has support?

Comment (smortex) [reply metadata.json:69]: According to https://www.suse.com/lifecycle 12.1 support ended on 2017-05-31 and 12.2 support ended on 2018-03-31.

Maybe we can adjust this list with `["12"]`?

Comment (bastelfreak) [reply metadata.json:69]: yes, we should switch to major versions only.

Comment (smortex) [reply metadata.json:70]: Updated in #1127

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**